sim/hunter/hunters_mark.go

```diff
@@ -0,0 +1,65 @@
+package hunter
+
+import (
+	"github.com/wowsims/sod/sim/core"
+)
+
+func (hunter *Hunter) getHuntersMark(rank int) core.SpellConfig {
+	spellId := [5]int32{0, 1130, 14323, 14324, 14325}[rank]
+	manaCost := [5]float64{0, 15, 30, 45, 60}[rank]
+	level := [5]int{0, 6, 22, 40, 58}[rank]
+
+	hunter.HuntersMarkAuras = hunter.NewEnemyAuraArray(hunter.myHuntersMarkAura)
+
+	return core.SpellConfig{
+		ClassSpellMask: ClassSpellMask_HunterHuntersMark,
+		ActionID:       core.ActionID{SpellID: spellId},
+		SpellSchool:    core.SpellSchoolArcane,
+		DefenseType:    core.DefenseTypeNone,
+		ProcMask:       core.ProcMaskEmpty,
+		Flags:          core.SpellFlagAPL,
+		Rank:           rank,
+		RequiredLevel:  level,
+
+		ManaCost: core.ManaCostOptions{
+			FlatCost: manaCost,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: core.GCDDefault,
+			},
+			IgnoreHaste: true, // Hunter GCD is locked at 1.5s
+		},
+
+		ThreatMultiplier: 0,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			aura := hunter.HuntersMarkAuras.Get(target)
+			// Can only have 1 hunter's mark up at a time
+			for _, auras := range hunter.HuntersMarkAuras {
+				if auras.IsActive() {
+					auras.Deactivate(sim)
+				}
+			}
+			aura.Activate(sim)
+		},
+
+		RelatedAuras: []core.AuraArray{hunter.HuntersMarkAuras},
+	}
+}
+
+func (hunter *Hunter) myHuntersMarkAura(target *core.Unit, playerLevel int32) *core.Aura {
+	return core.HuntersMarkAura(target, hunter.Talents.ImprovedHuntersMark, hunter.Level)
+}
```

Comment:
Hm thinking about this some more, this will make all of the ranks of hunters mark apply the same max available rank aura based on the player's level. Maybe we should just give access to the highest rank spell instead of having all ranks available? Is there ever a reason you'd want to downrank?